app/controllers/facility_accounts_controller.rb

```diff
@@ -79,13 +73,12 @@ def edit
 
   # PUT /facilities/:facility_id/accounts/:id
   def update
-    class_params = account_class_params
-
     if @account.is_a?(AffiliateAccount)
-      class_params[:affiliate_other] = nil if class_params[:affiliate_id] != Affiliate.OTHER.id.to_s
+      @account.affiliate_other = nil if params[:affiliate_id] != Affiliate.OTHER.id.to_s
     end
```

Comment:
Can someone explain how affiliates work exactly?  For example, `affiliate_other` vs `affiliate_id`.  And why are affiliates only supported on facility accounts?  

I think I successfully mimicked the existing affiliate functionality, but I can't with certainty that I fully understand affiliates.